### ISSUE TITLE
Voiding statements where the voided statement does not exist in the LRS leads to an error 400

### ISSUE THREAD

**Issue by garemoko** [LearningLocker/learninglocker#802]
**Version**
1.12.1

**Steps to reproduce the bug**

```
POST /data/xAPI/statements HTTP/1.1
Host: locker.watershedlrs.com
Authorization: Basic <auth details>
X-Experience-API-Version: 1.0.2
Content-Type: multipart/mixed; boundary=++++++
Cache-Control: no-cache
Postman-Token: d293cf5c-b24d-17cb-967e-851f29cddb00

--++++++
Content-Type:application/json

[
    {
        "id": "716be2c8-1b81-4a72-bf17-73ada6f6c576",
        "actor": {
            "mbox": "mailto:andrew.downes@watershedlrs.com",
            "objectType": "Agent"
        },
        "verb": {
            "id": "http://adlnet.gov/expapi/verbs/voided",
            "display": {
                "und": "voided"
            }
        },
        "timestamp": "2016-04-07T14:04:26.632Z",
        "stored": "2016-04-07T14:04:27.014Z",
        "authority": {
            "name": "Logstore",
            "account": {
                "homePage": "https://sandbox.watershedlrs.com",
                "name": "XmyWbsjFfQfwDL"
            },
            "objectType": "Agent"
        },
        "version": "1.0.0",
        "object": {
            "id": "95444d64-7d3f-4d5f-876a-78083043f16d",
            "objectType": "StatementRef"
        }
    }
]

--++++++--
```

**Expected behaviour**
No error should be returned; the statement should be stored. 

See (development) version 1.0.3 of xAPI [here](https://github.com/adlnet/xAPI-Spec/blob/1.0.3/xAPI-Data.md#232-voiding): 

```
Upon receiving a Statement that voids another, the LRS SHOULD NOT* reject the request on the grounds of the Object of that voiding Statement not being present.
```

There's a few reasons why this is important. 
**Actual behaviour**
Error 400 is returned. Content:

```
{
  "error": true,
  "success": false,
  "message": "xapi.errors.void_null",
  "code": 400,
  "trace": "#0 /var/www/learninglocker/app/locker/repository/Statement/EloquentVoider.php(19): Locker\\Repository\\Statement\\EloquentVoider->voidStatement(Object(stdClass), Object(Locker\\Repository\\Statement\\StoreOptions))\n#1 [internal function]: Locker\\Repository\\Statement\\EloquentVoider->Locker\\Repository\\Statement\\{closure}(Object(stdClass))\n#2 /var/www/learninglocker/app/locker/repository/Statement/EloquentVoider.php(20): array_map(Object(Closure), Array)\n#3 /var/www/learninglocker/app/locker/repository/Statement/EloquentStorer.php(41): Locker\\Repository\\Statement\\EloquentVoider->voidStatements(Array, Object(Locker\\Repository\\Statement\\StoreOptions))\n#4 /var/www/learninglocker/app/locker/repository/Statement/EloquentRepository.php(31): Locker\\Repository\\Statement\\EloquentStorer->store(Array, Array, Object(Locker\\Repository\\Statement\\StoreOptions))\n#5 /var/www/learninglocker/app/controllers/xapi/StatementStoreController.php(129): Locker\\Repository\\Statement\\EloquentRepository->store(Array, Array, Array)\n#6 /var/www/learninglocker/app/controllers/xapi/StatementStoreController.php(66): Controllers\\xAPI\\StatementStoreController->createStatements(Array)\n#7 /var/www/learninglocker/app/controllers/xapi/StatementController.php(73): Controllers\\xAPI\\StatementStoreController->store(Array)\n#8 /var/www/learninglocker/app/controllers/xapi/BaseController.php(45): Controllers\\xAPI\\StatementController->store()\n#9 [internal function]: Controllers\\xAPI\\BaseController->selectMethod()\n#10 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(231): call_user_func_array(Array, Array)\n#11 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(93): Illuminate\\Routing\\Controller->callAction('selectMethod', Array)\n#12 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(62): Illuminate\\Routing\\ControllerDispatcher->call(Object(Controllers\\xAPI\\StatementController), Object(Illuminate\\Routing\\Route), 'selectMethod')\n#13 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Routing/Router.php(967): Illuminate\\Routing\\ControllerDispatcher->dispatch(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request), 'Controllers\\xAP...', 'selectMethod')\n#14 [internal function]: Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()\n#15 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Routing/Route.php(109): call_user_func_array(Object(Closure), Array)\n#16 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Routing/Router.php(1033): Illuminate\\Routing\\Route->run(Object(Illuminate\\Http\\Request))\n#17 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Routing/Router.php(1001): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))\n#18 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(775): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))\n#19 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(745): Illuminate\\Foundation\\Application->dispatch(Object(Illuminate\\Http\\Request))\n#20 /var/www/learninglocker/vendor/itsgoingd/clockwork/Clockwork/Support/Laravel/ClockworkLegacyMiddleware.php(20): Illuminate\\Foundation\\Application->handle(Object(Illuminate\\Http\\Request), 1, true)\n#21 /var/www/learninglocker/vendor/asm89/stack-cors/src/Asm89/Stack/Cors.php(51): Clockwork\\Support\\Laravel\\ClockworkLegacyMiddleware->handle(Object(Illuminate\\Http\\Request), 1, true)\n#22 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Session/Middleware.php(72): Asm89\\Stack\\Cors->handle(Object(Illuminate\\Http\\Request), 1, true)\n#23 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Cookie/Queue.php(47): Illuminate\\Session\\Middleware->handle(Object(Illuminate\\Http\\Request), 1, true)\n#24 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Cookie/Guard.php(51): Illuminate\\Cookie\\Queue->handle(Object(Illuminate\\Http\\Request), 1, true)\n#25 /var/www/learninglocker/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Illuminate\\Cookie\\Guard->handle(Object(Illuminate\\Http\\Request), 1, true)\n#26 /var/www/learninglocker/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(641): Stack\\StackedHttpKernel->handle(Object(Illuminate\\Http\\Request))\n#27 /var/www/learninglocker/public/index.php(49): Illuminate\\Foundation\\Application->run()\n#28 {main}"
}
```

**Additional information**
The problem is the else clause here: https://github.com/LearningLocker/learninglocker/blob/develop/app/locker/repository/Statement/EloquentVoider.php#L40-L44

Removing that else clause resolves the issue.

**Comment by ryansmith94** [comment]
If that else clause is removed, does the voided statement actually get voided when it's sent afterwards?

**Comment by garemoko** [comment]
@ryansmith94 good question; I'm going to guess probably not! I imagine the fix for that is a little more involved so I'll leave this one with you guys. Selfishly, my copy is now working well enough for my particular purposes right now.

**Comment by ryansmith94** [comment]
Ok no worries thanks @garemoko

**Comment by ht2** [comment]
Addressed in v2.0.0
